src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -307,9 +309,19 @@ func (r *ClusterReconciler) Reconcile(
 	if err := r.setPodNodeIDAnnotation(ctx, &redpandaCluster, log); err != nil {
 		return ctrl.Result{}, fmt.Errorf("setting pod node_id annotation: %w", err)
 	}
-	if err := r.setLicense(ctx, &redpandaCluster, log); err != nil {
-		return ctrl.Result{}, fmt.Errorf("setting license: %w", err)
+
+	// want: refactor above to resources (i.e. setInitialSuperUserPassword, reconcileConfiguration)
+	// ensuring license must be at the end when condition ClusterConfigured=true and AdminAPI is ready
+	license := resources.NewLicense(r.Client, r.Scheme, &redpandaCluster, adminAPI, log)
+	if err := license.Ensure(ctx); err != nil {
+		var raErr *resources.RequeueAfterError
+		if errors.As(err, &raErr) {
+			log.Info(raErr.Error())
```

Comment:
Why log error as info?

Reply:
I put info same with the err checks above to be consistent.